plutus-ledger-api/src/Plutus/V1/Ledger/Api.hs

```diff
@@ -195,12 +196,12 @@ evaluateScriptCounting verbose costParams p args = swap $ runWriter @LogOutput $
     appliedTerm <- mkTermToEvaluate p args
     model <- liftEither $ mkCostModel costParams
 
-    let (res, final, logs) = UPLC.runCek
+    let (res, UPLC.CountingSt final, logs) = UPLC.runCek
           (PLC.toBuiltinsRuntime () model)
-          PLC.Counting
+          UPLC.counting
           (verbose == Verbose)
           appliedTerm
 
     tell $ Prelude.map Text.pack logs
     liftEither $ first CekError $ void res
-    pure $ PLC.toRequiredExBudget final
+    pure final
```

Comment:
Note that the interface hasn't really changed: the caller still specifies a budgeting mode (except now it starts with a lower-case letter rather than an upper-case one), but extracting the final state now does not require handling all possible state types -- just the one that is associated with the chosen mode (both `CountingSt` and `RestrictingSt` are `newtype`s, so we could just as well use `coerce`).